docs/orchestrating-elastic-stack-applications/logstash.asciidoc

```diff
@@ -837,15 +834,501 @@ spec:
+This could be for checkpointing to keep track of events already processed, a place to temporarily write events before sending a batch of events, or just to actually write events to disk in the case of {logstash-ref}/plugins-outputs-file.html[`logstash-output-file`].
+
+{ls} on ECK by default supplies a small 1.5 GiB (gibibyte) default persistent volume to each pod.
+This volume is called `logstash-data` and is located at `/usr/logstash/data`, and is typically the default location for most plugin use cases. 
+This volume is stable across restarts of {ls} pods and is suitable for many use cases.
+
+NOTE: When plugins use writable storage, each plugin must store its data a dedicated folder or file to avoid overwriting data.
+
+[id="{p}-logstash-working-with-plugins-writable-checkpointing"]
+===== Checkpointing
+
+Some {ls} plugins need to write "checkpoints" to local storage in order to keep track of events that have already been processed. 
+Plugins that retrieve data from external sources need to do this if the external source does not provide any mechanism to track state internally.
+
+Not all external data sources have mechanisms to track state internally, and {ls} checkpoints can help persist data. 
+
+In the plugin documentation, look for configurations that call for a `path` with a settings like `sincedb`, `sincedb_path`, `sequence_path`, or `last_run_metadata_path`. Check out specific plugin documentation in the {logstash-ref}[Logstash Reference] for details.
+
+[source, yaml,subs="attributes,+macros,callouts"]
+----
+spec:
+  pipelines:
+    - pipeline.id: main
+      config.string: |
+        input {
+          jdbc {
+             jdbc_driver_library => "/usr/share/logstash/jars/postgresql.jar"
+             jdbc_driver_class => "org.postgresql.Driver"
+             last_metadata_path => "/usr/share/logstash/data/main/logstash_jdbc_last_run <1>
+          }
+        }
+----
+<1> If you are using more than one plugin of the same type, specify a unique location for each plugin to use.
+
+If the default `logstash-data` volume is insufficient for your needs, see the volume section for details on how to add additional volumes.
+
+
+[id="{p}-logstash-working-with-plugins-writable-temp"]
+===== Writable staging or temporary data
+
+Some {ls} plugins write data to a staging directory or file before processing for input, or outputting to their final destination. 
+Often these staging folders can be persisted across restarts to avoid duplicating processing of data.
+
+In the plugin documentation, look for  names such as `tmp_directory`, `temporary_directory`, `staging_directory`.
+
+To persist data across pod restarts, set this value to point to the default `logstash-data` volume or your own PersistentVolumeClaim. 
+
+[source, yaml,subs="attributes,+macros,callouts"]
+----
+spec:
+  pipelines:
+    - pipeline.id: main
+      config.string: |
+        output {
+          s3 {
+             id => 'main_s3_output'
```

Comment:
```suggestion
             id => "main_s3_output"
```

Reply:
@robbavey GitHub didn't know how to resolve when I tried to apply two suggestions to the same line, and my ability to apply your suggestion went away. That's why I'm reintroducing your previous comment.